python/lsst/ts/mtdome/mock_llc/amcs.py

```diff
@@ -98,6 +156,170 @@ def __init__(self, start_tai: float) -> None:
         self.barcode_head_calibrated = np.zeros(_NUM_RESOLVERS, dtype=float)
         self.barcode_head_weighted = np.zeros(_NUM_RESOLVERS, dtype=float)
 
+        # State machine related attributes.
+        self.current_state = MotionState.PARKED.name
+        self.start_state = MotionState.PARKED.name
+        self.target_state = MotionState.PARKED.name
+
+        # Error state related attributes.
+        self.drives_in_error_state = [False] * NUM_MOTORS
+
+    async def evaluate_state(self, current_tai: float) -> None:
+        """Evaluate the state and perform a state transition if necessary.
+
+        Parameters
+        ----------
+        current_tai : `float`
+            The current time, in UNIX TAI seconds.
+        """
+        match self.target_state:
+            case (
+                MotionState.MOVING.name
+                | MotionState.CRAWLING.name
+                | MotionState.PARKED.name
+                | InternalMotionState.STATIONARY.name
+            ):
+                await self._handle_motion(current_tai)
+            case MotionState.STOPPED.name:
+                await self._handle_stopped(current_tai)
+            case _:
+                await self._warn_invalid_state()
+
+    async def _warn_invalid_state(self) -> None:
+        self.log.warning(f"Not handling invalid target state {self.target_state}")
```

Comment:
Function doc string? There are many functions in this PR do not have the doc string. Please consider to add them.